### PR TITLE
Begin modeling and exploration of GSVA scores

### PR DESCRIPTION
**Pull request by sjspielman**
### Purpose/implementation Section

To model and visualize GSVA scores.

#### What scientific question is your analysis addressing?

Which pathways show significant GSVA score differences across samples?

#### What was your approach?

A continued update of the now-deprecated `ssGSEA-hallmark` analysis, including:

1. Perform a series of ANOVA/Tukey tests to identify significant differences across pathways and histological types
2. _Briefly_, examine whether results are consistent between stranded and polyA expression data
3. Begin to devise a visualization strategy for exploring pathway differences for a given histology

#### What GitHub issue does your pull request address?

N/A

#### Which areas should receive a particularly close look?

1. Library loading is collapsed largely to tidyverse since the whole core is loaded anyways. Is this appropriate or should we be more explicit?

2. I'm unsure about the level of multiple testing correction. Since we are performing MANY posthoc Tukey tests, I have elected to just Bonferroni their P-values by dividing the Tukey P-values by the number of tests performed. This may or may not be reasonable. In addition, as separate tests were performed for polyA vs stranded expression data, the P-values used here are even more conservative.

3. Are the "discrepancies" between expression types reasonable? Seems to me, yes, given substantial differences in sample size between libraries.

4. Is the proposed plotting strategy reasonable? If so, can convert it to a function and extend its flexibility. 

#### Is there anything that you want to discuss further?

Are there other figures which would be helpful to include either a) as templates for others to make figures, or b) an aggressively _O M I C S_ figure?

3. Domain expertise

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Unclear to me, likely no.

### Results

Currently all initial exploration in notebook. This will come later.

#### Reproducibility Checklist

- [X ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [X ] This analysis has been added to continuous integration.
- [X ] This analysis is recorded in the table in `analyses/README.md`.

### CODE REVIEW

**Comment by sjspielman** [comment]
I naively suspect this fails CI due to the smaller data, given this error (there should always be two or more levels..):
```
Quitting from lines 84-127 (02-exploratory-gsea.Rmd) 
Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : 
  contrasts can be applied only to factors with 2 or more levels
Calls: <Anonymous> ... model.matrix -> model.matrix.default -> contrasts<-
```
Will investigate.

File: analyses/gene-set-enrichment-analysis/02-exploratory-gsea.Rmd [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Significance testing universal threshold
+SIGNIFICANCE_THRESHOLD   <- 0.01
+
+# Are we testing?
+if( !(params$is_ci %in% c(0,1) & !(params$is_ci) %in% c(TRUE, FALSE)) ){
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/gene-set-enrichment-analysis/02-exploratory-gsea.Rmd]: I think this CI failure is stemming from the same issue:

```
ERROR: The parameter `is_ci` should be 0/1 (or FALSE/TRUE).
```
So I will go ahead and commit this suggestion.

**Comment by sjspielman** [comment]
Great, thanks, some responses - 

>I am conditioned to look for a boxplot for this kind of thing, honestly. 

As in, X-axis as pathway and Y-axis as GSVA scores for all samples? Should I make a template and/or replace barplots?

>I also think it is reasonable to include a heatmap, which you don't have to do in this pull request.

Yes, will do next to wrap up Issue #7 

----
Also, will we want to perform anova's other than looking across `short_histology`? I can make the code more flexible as needed.

**Comment by jaclyn-taroni** [comment]
> As in, X-axis as pathway and Y-axis as GSVA scores for all samples? Should I make a template and/or replace barplots?

I was thinking X-axis for `short_histology` (or whatever else) and Y-axis is GSVA scores, where you plot a single pathway at a time, or I can imagine a situation where you filter to 5(ish) pathways of interest and then use the pathways as facets.

>  Also, will we want to perform anova's other than looking across `short_histology`? I can make the code more flexible as needed.

Yes, I suspect we may want to look at `disease_type_new` for example. I would also probably keep in mind that we may want to subset to a particular histology and then compare between molecular subtypes as you're implementing things.

**Pull request merged**